portfolio/src/main/webapp/src/com.js

```diff
@@ -43,12 +43,49 @@ new MDCRipple(document.querySelector('.mdc-fab'));
 /**
  * Gets comments data from server by submitting a GET
  * request to /data. Returns the comments as a JavaScript
- * value promise.
+ * value promise. maxComments constrains the number of comments
+ * returned by the server, provided it is a valid integer.
+ * Otherwise, all comments are returned.
  *
+ * @param {String} maxComments
  * @return {Promise<any>}
  */
-function getCommentsFromServer() {
-  return fetch('/data').then((response) => response.json());
+function getCommentsFromServer(maxComments) {
+  // Client-side argument validation. This is done
+  // server-side as well.
+  let argString = '';
+  // Positive Integer
+  if (/(\+)?[0-9]+/.test(maxComments)) {
```

Comment:
is this necessary considering that `maxComments` is coming from our own component? It isn't random user input (i.e. it isn't coming from a text box, its a dropdown).